test/ibmq/test_basic_server_paths.py

```diff
@@ -0,0 +1,117 @@
+
+import time
+
+from qiskit.test import slow_test
+from qiskit.providers.jobstatus import JOB_FINAL_STATES, JobStatus
+
+from qiskit.providers.ibmq import least_busy
+from ..decorators import requires_providers
+from ..ibmqtestcase import IBMQTestCase
+from ..utils import cancel_job, bell_in_qobj
+
+
+class TestBasicServerPaths(IBMQTestCase):
+    """Test the basic server endpoints using both a public and premium provider."""
+
+    @classmethod
+    @requires_providers
+    def setUpClass(cls, providers):
+        # pylint: disable=arguments-differ
+        super().setUpClass()
+        cls.providers = providers  # Dict[str, AccountProvider]
+
+    @slow_test
+    def test_job_submission(self):
+        """Test running a job against a device."""
+        for _desc, provider in self.providers.items():
+            backend = least_busy(provider.backends(
+                simulator=False, filters=lambda b: b.configuration().n_qubits >= 5))
+            provider_backend = {'provider': provider, 'backend': backend}
+            with self.subTest(provider_backend=provider_backend):
+                qobj = bell_in_qobj(backend)
+                job = backend.run(qobj, validate_qobj=True)
+
+                # Fetch the results.
+                result = job.result()
+                self.assertTrue(result.success)
+
+                # Fetch the qobj.
+                qobj_downloaded = job.qobj()
+                self.assertEqual(qobj_downloaded, qobj.to_dict())
+
+    def test_job_backend_properties_and_status(self):
+        """Test the backend properties and status of a job."""
+        for _desc, provider in self.providers.items():
+            backend = provider.backends(
+                simulator=False, operational=True,
+                filters=lambda b: b.configuration().n_qubits >= 5)[0]
+            provider_backend = {'provider': provider, 'backend': backend}
+            with self.subTest(provider_backend=provider_backend):
+                qobj = bell_in_qobj(backend)
+                job = backend.run(qobj, validate_qobj=True)
+
+                self.assertIsNotNone(job.properties())
+                self.assertTrue(job.status())
+                # Cancel job so it doesn't consume more resources.
+                cancel_job(job, verify=True)
+
+    def test_retrieve_jobs(self):
+        """Test retrieving jobs."""
+        backend_name = 'ibmq_qasm_simulator'
+        for _desc, provider in self.providers.items():
+            with self.subTest(provider=provider):
+                backend = provider.get_backend(backend_name)
+                qobj = bell_in_qobj(backend)
+
+                job = backend.run(qobj, validate_qobj=True)
+                job_id = job.job_id()
+
+                # TODO No need to wait for job to run once api is fixed
+                while job.status() not in JOB_FINAL_STATES + (JobStatus.RUNNING,):
+                    time.sleep(0.5)
```

Comment:
This is only an issue if you're querying for a specific job status (which you're not)